mempool/src/shared_mempool/priority.rs

```diff
@@ -529,6 +556,60 @@ fn compare_ping_latency(
     }
 }
 
+/// Returns whether the given peer monitoring metadata is healthy. A peer is
+/// considered healthy if its latest ledger timestamp is within the max acceptable
+/// sync lag. If the monitoring metadata is missing, then we return None.
+fn check_peer_metadata_health(
+    mempool_config: &MempoolConfig,
+    time_service: &TimeService,
+    monitoring_metadata: &Option<&PeerMonitoringMetadata>,
+) -> Option<bool> {
+    monitoring_metadata.and_then(|metadata| {
+        metadata
+            .latest_node_info_response
+            .as_ref()
+            .map(|node_information_response| {
+                // Get the peer's ledger timestamp and the current timestamp
+                let peer_ledger_timestamp_usecs = node_information_response.ledger_timestamp_usecs;
+                let current_timestamp_usecs = get_timestamp_now_usecs(time_service);
+
+                // Calculate the max sync lag before the peer is considered unhealthy (in microseconds)
+                let max_sync_lag_secs = mempool_config.max_sync_lag_before_unhealthy_secs as u64;
+                let max_sync_lag_usecs = max_sync_lag_secs * MICROS_PER_SECOND;
+
+                // Determine if the peer is healthy
+                current_timestamp_usecs.saturating_sub(peer_ledger_timestamp_usecs)
+                    < max_sync_lag_usecs
+            })
+    })
+}
+
+/// Compares the health of the given peer monitoring metadata. Healthy
+/// peers are prioritized over unhealthy peers, or peers missing metadata.
+/// If a peer is unhealthy, and another peer is missing metadata, they are
+/// considered equal.
+fn compare_peer_health(
+    mempool_config: &MempoolConfig,
+    time_service: &TimeService,
+    monitoring_metadata_a: &Option<&PeerMonitoringMetadata>,
+    monitoring_metadata_b: &Option<&PeerMonitoringMetadata>,
+) -> Ordering {
+    // Check the health of the peer monitoring metadata
+    let is_healthy_a =
+        check_peer_metadata_health(mempool_config, time_service, monitoring_metadata_a);
+    let is_healthy_b =
+        check_peer_metadata_health(mempool_config, time_service, monitoring_metadata_b);
+
+    // Compare the health statuses
+    match (is_healthy_a, is_healthy_b) {
+        (Some(true), Some(false)) => Ordering::Greater, // A is healthy, B is unhealthy
+        (Some(false), Some(true)) => Ordering::Less,    // A is unhealthy, B is healthy
+        (Some(true), None) => Ordering::Greater,        // A is healthy, B is missing data
+        (None, Some(true)) => Ordering::Less,           // A is missing data, B is healthy
+        _ => Ordering::Equal, // Both are healthy, unhealthy or missing data
```

Comment:
doesn't look like we need an Option here? seems just unwrap_or(false) is good enough?